src/TributeMenuEvents.js

```diff
@@ -6,32 +6,47 @@ class TributeMenuEvents {
     }
 
     bind(menu) {
-        menu.addEventListener('keydown',
-            this.tribute.events.keydown.bind(this.menu, this), false)
-        this.tribute.range.getDocument().addEventListener('mousedown',
-            this.tribute.events.click.bind(null, this), false)
-        window.addEventListener('resize', this.debounce(() => {
+        this.menuKeydownEvent = this.tribute.events.keydown.bind(this.menu, this)
+        this.menuClickEvent = this.tribute.events.click.bind(null, this)
+        this.menuContainerScrollEvent = this.debounce(() => {
+            if (this.tribute.isActive) {
+                this.tribute.showMenuFor(this.tribute.current.element, false)
+            }
+        }, 300, false)
+        this.windowResizeEvent = this.debounce(() => {
             if (this.tribute.isActive) {
                 this.tribute.range.positionMenuAtCaret(true)
             }
-        }, 300, false))
+        }, 300, false)
+
+        menu.addEventListener('keydown',
+            this.menuKeydownEvent, false)
+        this.tribute.range.getDocument().addEventListener('mousedown',
+            this.menuClickEvent, false)
+        window.addEventListener('resize', this.windowResizeEvent)
 
         if (this.menuContainer) {
-            this.menuContainer.addEventListener('scroll', this.debounce(() => {
-                if (this.tribute.isActive) {
-                    this.tribute.showMenuFor(this.tribute.current.element, false)
-                }
-            }, 300, false), false)
+            this.menuContainer.addEventListener('scroll', this.menuContainerScrollEvent, false)
         } else {
-            window.onscroll = this.debounce(() => {
-                if (this.tribute.isActive) {
-                    this.tribute.showMenuFor(this.tribute.current.element, false)
-                }
-            }, 300, false)
+            window.addEventListener('scroll', this.menuContainerScrollEvent)
         }
 
     }
 
+    unbind(menu) {
+        menu.removeEventListener('keydown',
```

Comment:
In my usage scenario, after the user opens the input box, it does not necessarily use '@'. At this time, if I use 'detach', an error occurs because the menu is 'undefined' at this time.
```
if (menu) {
  menu.removeEventListener('keydown', this.menuKeydownEvent, false);
}
```